File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.cpp
// Last updated: 2/11/2026, 11:08:24 AM
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int>left (nums.size(), 1);
        vector<int>right (nums.size(),1);

        int leftproduct = 1;
        for (int i = 0; i < nums.size(); ++i){
            left[i] = leftproduct;
            leftproduct *= nums[i];
        }

        int rightproduct = 1;
        for (int i = nums.size() -1; i >= 0; --i){
            right[i] = rightproduct;
            rightproduct *= nums[i];
        }

        vector<int> res (nums.size());

        for (int i = 0; i<nums.size(); i++){
            res[i] = right[i] * left[i];
        }
        return res;
    }
};

File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 2/11/2026, 11:08:28 AM
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [1]* len(nums)
        right = [1] * len(nums)

        leftproduct = 1
        for i in range(len(nums)):
            left[i] = leftproduct
            leftproduct *= nums[i]

        rightproduct = 1
        for i in range(len(nums)-1, -1,-1):
            right[i] = rightproduct
            rightproduct *= nums[i]

        res = [0] * len(nums)
        for i in range(len(nums)):
            res[i] = left[i] * right[i]

        return res
